### PR TITLE
[CMTOOL-309] support for migrations from EAP 7.x to EAP8 GA

### PR DESCRIPTION
**Pull request by emmartins**
Issue: https://issues.redhat.com/browse/CMTOOL-309

### CODE REVIEW

File: servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/LegacySecurityConfiguration.java [hunk 1]
```diff
@@ -64,6 +64,50 @@ public void setDomainControllerRemoteSecurityRealm(String domainControllerRemote
...
+    public String getDefaultElytronManagementRealmName() {
+        return "migration-defaultManagementRealm";
+    }
+
+    public String getDefaultElytronApplicationDomainName() {
+        return "migration-defaultApplicationDomain";
+    }
+
+    public String getDefaultElytronManagementDomainName() {
+        return "migration-defaultManagemenDomain";
```
Comment (OndrejKotek) [inline servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/LegacySecurityConfiguration.java:80]: Typo, missing "t": migration-defaultManagemenDomain -> migration-defaultManagementDomain

File: migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                .domain(serverUpdateBuilders.domainBuilder()
+                        .domainConfigurations(serverUpdateBuilders.domainConfigurationBuilder()
+                                .subtask(new WildFly27_0MigrateJBossDomainProperties<>())
+                                .subtask(legacySecurityConfigurationMigration.getReadLegacySecurityConfiguration())
+                                .subtask(new RemoveUnsupportedExtensions<>())
+                                .subtask(new RemoveUnsupportedSubsystems<>())
+                                .subtask(new MigrateReferencedModules<>())
+                                .subtask(new WildFly26_0MigrateReferencedPaths<>())
+                                .subtask(legacySecurityConfigurationMigration.getRemoveLegacySecurityRealms())
+                                .subtask(new EAP8_0AddHostExcludes<>())
```
Comment (OndrejKotek) [inline migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java:83]: This task may need some adjustments: IIRC we support just EAP 7.4 in domain of EAP 8.0, but there are also exclusions for EAP 6.2 - 7.3. Furthermore, for EAP 7.4 there is exclusion for `opentelemetry` subsystem which is not in the default configuration.

Do you want me to create a CMTOOL issue?

Comment (OndrejKotek) [reply migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java:83]: Created https://issues.redhat.com/browse/CMTOOL-340

File: servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/MigrateLegacySecurityRealmsToElytron.java [hunk 3]
```diff
@@ -78,172 +92,219 @@ protected MigrateToElytron(final LegacySecurityConfigurations legacySecurityConf
...
-            if (securityRealm.getAuthentication() != null && securityRealm.getAuthorization() != null) {
-                addSecurityRealm(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addSecurityDomain(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addHttp(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addSasl(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-            }
-            addServerIdentities(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
+        protected void addOperationSteps(LegacySecurityConfiguration legacySecurityConfiguration, SubsystemResource subsystemResource, Operations.CompositeOperationBuilder compositeOperationBuilder, TaskContext taskContext) {
```
Comment (OndrejKotek) [inline servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/MigrateLegacySecurityRealmsToElytron.java:118]: The default resources are added even when not needed.

File: servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/LegacySecurityConfiguration.java (hunk 1 above)
Comment (emmartins) [reply servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/LegacySecurityConfiguration.java]: fixed

File: migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java (hunk 2 above)
Comment (emmartins) [reply migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java:83]: Thanks for the JIRA, let's sync this task ops later later there, once we know the final host excludes config, which frequently has "last minute" changes

File: servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/MigrateLegacySecurityRealmsToElytron.java (hunk 3 above)
Comment (emmartins) [reply servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/MigrateLegacySecurityRealmsToElytron.java:118]: This is intentional, I thought about it and it requires a lot of logic at XML level to figure out when it's needed or not (e.g. missing attributes which defaults to a legacy security realm or domain). Those resources need to be explicitly referenced to be used so I don't think there is any issue with always creating them. If you don't agree please open a new bug JIRA.

**Pull request merged**